src/targets/maven.ts

```diff
@@ -1,43 +1,38 @@
 import { TargetConfig } from '../schemas/project_config';
-import { MavenReleaser } from '../utils/mavenDeployment';
 import { BaseArtifactProvider } from '../artifact_providers/base';
 import { BaseTarget } from './base';
 import { ConfigurationError } from '../utils/errors';
-import { withTempDir } from '../utils/files';
-import { isDryRun } from '../utils/helpers';
-import * as Github from '@octokit/rest';
-import {
-  getAuthUsername,
-  getGithubApiToken,
-  getGithubClient,
-  GithubRemote,
-} from '../utils/githubApi';
-import simpleGit from 'simple-git';
 import { homedir } from 'os';
-import { join } from 'path';
+import * as path from 'path';
 import * as fs from 'fs';
+import { exec } from 'child_process';
+import { isDryRun } from '../utils/helpers';
+import { withRetry } from '../utils/async';
 
-// TODO: add docs to the readme
-
-const GIT_REPO_OWNER = 'getsentry';
-const GIT_REPO_NAME = 'sentry-java';
 const GRADLE_PROPERTIES_FILENAME = 'gradle.properties';
-const FILES_TO_COMMIT = [GRADLE_PROPERTIES_FILENAME];
+
+const ANDROID_DIST_EXTENSION = '.aar'; // Must include the leading `.`
+const ANDROID_RELEASE_SUBSTR = 'release';
 
 /** Config options for the "maven" target. */
-interface MavenTargetConfig {
-  // Required env variables
-  ossrhUsername: string; // OSSRH_USERNAME
-  ossrhPassword: string; // OSSRH_PASSWORD
-  mavenUsername: string; // MAVEN_CENTRAL_USERNAME
-  mavenPassword: string; // MAVEN_CENTRAL_PASSWORD
-  // Optional env variables (have a default value)
-  distributionsPath: string; // DISTRIBUTIONS_PATH
-  settingsPath: string; // SETTINGS_PATH
-  mavenRepoUrl: string; // MAVEN_REPO_URL
-  mavenRepoId: string; // MAVEN_REPO_ID
-  mavenCliPath: string; // MAVEN_CLI_PATH
-  gradleCliPath: string; // GRADLE_CLI_PATH
+type MavenTargetConfig = {
+  [key in keyof (typeof RequiredConfig & typeof OptionalConfig)]: string;
+};
+
+enum RequiredConfig {
+  ossrhUsername = 'OSSRH_USERNAME',
+  ossrhPassword = 'OSSRH_PASSWORD',
+  mavenUsername = 'MAVEN_CENTRAL_USERNAME',
+  mavenPassword = 'MAVEN_CENTRAL_PASSWORD',
+}
```

Comment:
Just make these a `Set`?
```suggestion
const REQUIRED_ENV_CONFIG = new Set([
  'OSSRH_USERNAME',
  'OSSRH_PASSWORD',
  'MAVEN_CENTRAL_USERNAME',
  'MAVEN_CENTRAL_PASSWORD',
]);
```

Reply:
Why? Using an `enum` allows to not hardcode the variable multiple times, e.g. [here](https://github.com/getsentry/craft/blob/48380191fb7551311e53cca541e89d5784dc15f3/src/targets/maven.ts#L71-L84); and to have stronger typing (instead of weaker types and checking afterward), e.g. [here](https://github.com/getsentry/craft/blob/48380191fb7551311e53cca541e89d5784dc15f3/src/targets/maven.ts#L88).